### PR TITLE
add asurl detection to image output

### PR DESCRIPTION
**Pull request by amorin-gladia**
Fix (with [api](https://github.com/gladiaio/api/pull/143) and [office](https://github.com/gladiaio/office-addin/pull/91) PR) the excel bug on background removal and image-generation

### CODE REVIEW

**Comment by kuroidoruido** [comment]
"asUrl" query param or header has always been a dirty hack. There is a standard way to do the same thing: explicitly passing the header "Accept: application/json" and handle this header correctly in the api. I say it multiple time using "asurl" is the worst way to doing it!

**Review by jsoto-gladia — APPROVED** [review]

**Comment by amorin-gladia** [comment]
> "asUrl" query param or header has always been a dirty hack. There is a standard way to do the same thing: explicitly passing the header "Accept: application/json" and handle this header correctly in the api. I say it multiple time using "asurl" is the worst way to doing it!

@kuroidoruido  As it is with this PR, use "Accept: application/json"  is actually working to parse the result as a JSON.
The difference is that we can also use a specific arg in param. It is usefull to do it in order to store the response in cache correctly. It's also better for differentiate image-generation arrayBuffer parsing from image-generation URL array, with are both JSON, but not both asUrl. Just to demonstate that asUrl and accept:JSON are not covering exactly the same info

**Comment by kuroidoruido** [comment]
> > "asUrl" query param or header has always been a dirty hack. There is a standard way to do the same thing: explicitly passing the header "Accept: application/json" and handle this header correctly in the api. I say it multiple time using "asurl" is the worst way to doing it!
> 
> @kuroidoruido As it is with this PR, use "Accept: application/json" is actually working to parse the result as a JSON. The difference is that we can also use a specific arg in param. It is usefull to do it in order to store the response in cache correctly. It's also better for differentiate image-generation arrayBuffer parsing from image-generation URL array, with are both JSON, but not both asUrl. Just to demonstate that asUrl and accept:JSON are not covering exactly the same info

I am not agree. asUrl will convert the image given by the aipi backend to an url given as a json by uploading the image somewhere to have an url.

example:
```bash
curl -X 'POST' \
  'https://v2-api.gladia.io/text/image/image-generation/?model=stable-diffusion--stabilityai-sd-21' \
  -H 'x-gladia-key: token_1' \
  -H 'Content-Type: application/x-www-form-urlencoded' -H 'asurl: true' \
  -d 'prompt=The%20Eiffel%20tower%20landing%20on%20the%20moon%2C%20unreal%20engine.&samples=1&steps=40&seed=424242'
  
{"url":["https://yuyu-test-yuyu.s3.us-west-2.amazonaws.com/95336228.jpg"]}
```

So there is no difference between "Accept: application/json" and "asurl: true" in term of logic. If if you replace `-H 'asurl: true'` with `-H 'Accept: application/json'` the api should return the image(s) in json format.

To be perfect we should have:
- if we have samples > 1: we keep the actual behavior and return a json array with data urls
- if we have samples === 1: we return directly the image (does not work, I get a Internal Server Error with swagger or curl)
- if we have samples === 1 && Accept === 'application/json': we convert the image as data url to return it with the same format as for multiple images (does not work, I get a Internal Server Error with swagger or curl)

This way:
- the behavior is standard
- the behavior is working for every kind of task
- we do not need to store anywhere the image

**Comment by amorin-gladia** [comment]
@kuroidoruido I modified consequently (exept for data url, it is stored in external serveur in order to create url).
So no more param, only header now.
But I still prefer my solution, it forces me to use more complex mecanism to know if I have url or not for the cache (when the ouput is text, accept header is also json, so json !== url).

File: src/internal/http-client/xhr.http-client.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const XhrHttpClient: HttpClientFactory = ({ baseHeaders, baseUrl, httpCli
       return new Promise((resolve, reject) => {
         const xhr = new XMLHttpRequest();
         xhr.onload = () => {
-          if (responseType === 'json') {
+          if (responseType.includes('json')) {
```
Comment (kuroidoruido) [inline src/internal/http-client/xhr.http-client.ts:16]: no need for ".includes" here, because we want the responseType to be exactly "json"

**Review by kuroidoruido — CHANGES_REQUESTED** [review]

**Review by kuroidoruido — APPROVED** [review]

**Pull request merged**